cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGeneratorTest.java

```diff
@@ -158,27 +161,72 @@ public FileUploadHandler(File outputDir) {
     public HttpContentConsumer upload(HttpServiceRequest request,
                                       HttpServiceResponder responder,
                                       @PathParam("file") String file) throws IOException {
-      final WritableByteChannel channel = Channels.newChannel(new FileOutputStream(new File(outputDir, file)));
-      return new HttpContentConsumer() {
-        @Override
-        public void onReceived(ByteBuffer chunk, Transactional transactional) throws Exception {
-          channel.write(chunk);
-        }
+      return new FileContentConsumer(new File(outputDir, file));
+    }
 
-        @Override
-        public void onFinish(HttpServiceResponder responder) throws Exception {
-          channel.close();
-          responder.sendStatus(200);
-        }
+    @Path("/download/{file}")
+    @GET
+    public void download(HttpServiceRequest request,
+                         HttpServiceResponder responder,
+                         @PathParam("file") String file) {
+      Location location = Locations.toLocation(new File(outputDir, file));
+      try {
+        responder.send(200, location, "text/plain");
```

Comment:
We can provide something like `PartitionedFIleSetContentConsumer` in cdap-api to do so.